### PR TITLE
implemented tamagotchiStates

### PR DESCRIPTION
**Pull request by Adriann77**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,14 +1,18 @@
 import Game from './js/game.js';
-
+import TamagotchiStates from './js/modules/tamagotchiStates.js';
 
 document.addEventListener('DOMContentLoaded', () => {
+	const currentState = document.querySelector('.rectangle')
 	const game = new Game();
+	const tamaStates = new TamagotchiStates(currentState);
```
Comment (Sumick) [inline app.js]: **nitpick:** You don't need to create new class - let's put this logic inside `Tamagotchi` class. Or even if you'd like to leave this class, it should be "injected" into Tamagotchi class anyway

File: js/modules/tamagotchiStates.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+export default class TamagotchiStates {
+	constructor(status) {
+		this.status = status;
```
Comment (Sumick) [inline js/modules/tamagotchiStates.js]: **praise:** It's good that you keep state this way

**Review by Sumick — CHANGES_REQUESTED** [review]
You are really close! Let's improve Tamagotchi class composition to make it better in terms of maintainability

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
Based on current Tamagotchi state, display specific images sprite (or single image).

### Requirements
1. Should have only one state at the time based life parameter importance
2. Should be `happy` when all parameters are higher than or equal to 7
3. Should be `sad` when `fun` parameter is below or equal to 6
4. Should be `hungry` when `hunger` is below or equal to 6
5. Should be `sleepy` when `energy` is below or equal to 6

### Review

The code changes seem to cover the task requirements. However, there are a few issues that need to be addressed:

1. The code does not respect the importance of life parameters. According to the requirements, the Tamagotchi should have only one state at a time based on life parameter importance. However, the current implementation does not take this into account. For example, if `fun` (importance 4) is below 7 and `hunger` (importance 3) is also below 7, the Tamagotchi will be in the `hungry` state, even though `fun` has a higher importance.

2. The code does not handle the `sad` state correctly. According to the requirements, the Tamagotchi should be `sad` when the `fun` parameter is below or equal to 6. However, the current implementation sets the state to `bored`, not `sad`.

Here's how you can fix these issues:

```diff
checkCurrentStatus() {
    const tamagoSpirit = document.querySelector('.rectangle');
    const tamagoText = document.querySelector('.rectangle__status--text');
    tamagoSpirit.classList.remove(tamagoSpirit.classList[1]);

+   const states = [
+       { value: this.fun.value, state: 'sad', importance: this.fun.importance },
+       { value: this.hunger.value, state: 'hungry', importance: this.hunger.importance },
+       { value: this.energy.value, state: 'sleepy', importance: this.energy.importance },
+   ];
+   const currentState = states.sort((a, b) => b.importance - a.importance).find(state => state.value <= 6);
+
    if (this.health.value >= 7 && this.hunger.value >= 7 && this.fun.value >= 7 && this.energy.value >= 7) {
        tamagoSpirit.classList.add('happy');
        tamagoText.textContent = tamagoSpirit.classList[1].toUpperCase();
-   } else if (this.energy.value <= 6) {
-       tamagoSpirit.classList.add('sleepy');
-       tamagoText.textContent = tamagoSpirit.classList[1].toUpperCase();
-   } else if (this.hunger.value <= 6) {
-       tamagoSpirit.classList.add('hungry');
-       tamagoText.textContent = tamagoSpirit.classList[1].toUpperCase();
-   } else if (this.fun.value <= 6) {
-       tamagoSpirit.classList.add('bored');
-       tamagoText.textContent = 'SAD';
+   } else if (currentState) {
+       tamagoSpirit.classList.add(currentState.state);
+       tamagoText.textContent = currentState.state.toUpperCase();
    }
}
```

This code creates an array of possible states and sorts them by importance. It then finds the first state where the value is below or equal to 6. This ensures that the Tamagotchi's state is determined by the most important life parameter that is not in a good state.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
Based on current Tamagotchi state, display specific images sprite (or single image).

### Requirements
1. Should have only one state at the time based life parameter importance
2. Should be `happy` when all parameters are higher than or equal to 7
3. Should be `sad` when `fun` parameter is below or equal to 6
4. Should be `hungry` when `hunger` is below or equal 6
5. Should be `sleepy` when `energy` is below or equal 6

### Code Changes Review

#### app.js
The changes in this file are correct. The Tamagotchi class is now being exported, which is necessary for other modules to use it. The properties of the Tamagotchi class are correctly initialized according to the task requirements.

#### css/styles.css & sass/_spritesStates.scss
The changes in these files are correct. The class name has been changed from `.bored` to `.sad` to match the task requirements.

#### js/game.js
The changes in this file are correct. The console.log statement has been removed, which is a good practice as it cleans up the console output.

#### js/modules/tamagotchi.js
The changes in this file partially meet the task requirements. The `checkCurrentStatus` method correctly checks the current state of the Tamagotchi and updates the class and text content of the Tamagotchi element accordingly. However, there are a couple of issues:

1. The `checkCurrentStatus` method does not correctly handle the case when the `fun` parameter is below or equal to 6. According to the task requirements, the Tamagotchi should be `sad` in this case, but the current implementation will not set the Tamagotchi to `sad` unless `fun` has the highest importance and its value is the lowest. This could be fixed by adding a separate condition for `fun` before sorting the states array.

2. The `checkCurrentStatus` method does not correctly handle the case when the `hunger` parameter is below or equal to 6. According to the task requirements, the Tamagotchi should be `hungry` in this case, but the current implementation will not set the Tamagotchi to `hungry` unless `hunger` has the highest importance and its value is the lowest. This could be fixed by adding a separate condition for `hunger` before sorting the states array.

3. The `checkCurrentStatus` method does not correctly handle the case when the `energy` parameter is below or equal to 6. According to the task requirements, the Tamagotchi should be `sleepy` in this case, but the current implementation will not set the Tamagotchi to `sleepy` unless `energy` has the highest importance and its value is the lowest. This could be fixed by adding a separate condition for `energy` before sorting the states array.

Here is a code example that fixes these issues:

```javascript
checkCurrentStatus() {
    const tamagoSpirit = document.querySelector('.rectangle');
    const tamagoText = document.querySelector('.rectangle__status--text');
    tamagoSpirit.classList.remove(tamagoSpirit.classList[1]);

    if (this.fun.value <= 6) {
        tamagoSpirit.classList.add('sad');
        tamagoText.textContent = 'SAD';
    } else if (this.hunger.value <= 6) {
        tamagoSpirit.classList.add('hungry');
        tamagoText.textContent = 'HUNGRY';
    } else if (this.energy.value <= 6) {
        tamagoSpirit.classList.add('sleepy');
        tamagoText.textContent = 'SLEEPY';
    } else if (this.health.value >= 7 && this.hunger.value >= 7 && this.fun.value >= 7 && this.energy.value >= 7) {
        tamagoSpirit.classList.add('happy');
        tamagoText.textContent = 'HAPPY';
    } else {
        const states = [
            { value: this.hunger.value, importance: this.hunger.importance, state: 'HUNGRY' },
            { value: this.energy.value, importance: this.energy.importance, state: 'SLEEPY' },
            { value: this.fun.value, importance: this.fun.importance, state: 'SAD' },
        ];

        const currentState = states.sort((a, b) => a.importance - b.importance)[0];
        tamagoSpirit.classList.add(currentState.state.toLowerCase());
        tamagoText.textContent = currentState.state;
    }
}
```

This code first checks if `fun`, `hunger`, or `energy` is below or equal to 6 and sets the Tamagotchi to `sad`, `hungry`, or `sleepy` accordingly. If none of these conditions are met, it then checks if all parameters are higher than or equal to 7 and sets the Tamagotchi to `happy` if they are. If none of these conditions are met, it sorts the states array by importance and sets the Tamagotchi to the state with the highest importance and lowest value.

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task goal
Based on current Tamagotchi state, display specific images sprite (or single image).

### Requirements
1. Should have only one state at the time based life parameter importance.
2. Should be `happy` when all parameters are higher than or equal to 7.
3. Should be `sad` when `fun` parameter is below or equal to 6.
4. Should be `hungry` when `hunger` is below or equal 6.
5. Should be `sleepy` when `energy` is below or equal 6.

### Code Changes Review

#### app.js
The changes in this file are mostly formatting and do not affect the functionality of the code. No issues found.

#### css/styles.css
The changes in this file add transitions to the Tamagotchi's SVG and status bar. This doesn't directly relate to the task requirements, but it could improve the user experience by making state changes smoother. No issues found.

#### js/game.js
The changes in this file remove a console log statement. This doesn't directly relate to the task requirements, but it's a good practice to remove console logs in production code. No issues found.

#### js/modules/tamagotchi.js
The changes in this file add a new method `checkCurrentStatus` which checks the current state of the Tamagotchi based on its parameters and updates the class and text of the Tamagotchi's element accordingly. This method is called every second in the `mount` method. This covers all task requirements.

However, there's a potential issue: if two or more parameters are below or equal to 6, the state with the highest importance will be chosen, but if two or more states have the same importance, only the first one found will be chosen. This could lead to inconsistent behavior if the importance of the states is not carefully managed.

#### sass/_spritesStates.scss and sass/styles.scss
The changes in these files are similar to those in css/styles.css and do not directly relate to the task requirements. No issues found.

### Conclusion
The code changes cover all task requirements. However, there's a potential issue with how the states are chosen when multiple parameters are below or equal to 6. This could be improved by adding a mechanism to handle such cases, for example by choosing the state of the parameter that is the lowest.

**Pull request merged**